core/src/main/scala/Server.scala

```diff
@@ -57,8 +61,8 @@ object Server {
           val deltaNanos = System.nanoTime - startNanos
           val delta = Duration.fromNanos(deltaNanos)
           val result = right(a)
-          monitoring.handled(ctx, r, expected, result, delta)
-          toTask(codecs.responseEncoder(respEncoder).encode(result))
+          monitoring.handled(ctx, r, expected, right(a), delta)
```

Comment:
@jedesah I forgot to use result in handled, it would be great if you could update the code :-)